File: app/src/main/java/de/westnordost/streetcomplete/data/VisibleQuestListener.java
package de.westnordost.streetcomplete.data;

import java.util.Collection;

public interface VisibleQuestListener
{
	void onQuestsCreated(Collection<? extends Quest> quests, QuestGroup group);
	/** Called when the given quests are removed I.e. solved, hidden by the user or when they become obsolete. */
	void onQuestsRemoved(Collection<Long> questIds, QuestGroup group);
}

File: app/src/main/java/de/westnordost/streetcomplete/data/VisibleQuestListener.kt
package de.westnordost.streetcomplete.data

import de.westnordost.streetcomplete.data.quest.Quest
import de.westnordost.streetcomplete.data.quest.QuestGroup

interface VisibleQuestListener {
    fun onQuestsCreated(quests: Collection<Quest>, group: QuestGroup)
    /** Called when the given quests are removed I.e. solved, hidden by the user or when they become obsolete.  */
    fun onQuestsRemoved(questIds: Collection<Long>, group: QuestGroup)
}
